test/test_distributions.py

```diff
@@ -249,6 +249,31 @@ def fn(args):
         assert_allclose(np.sum(actual_grad[i]), expected_grad, rtol=0.02)
 
 
+@pytest.mark.parametrize('jax_dist, sp_dist, params', [
+    (dist.Gamma, osp.gamma, (1.,)),
+    (dist.Gamma, osp.gamma, (0.1,)),
+    (dist.Gamma, osp.gamma, (10.,)),
+    # TODO: add more test cases for Beta/StudentT (and Dirichlet too) when
+    # their pathwise grad (independent of standard_gamma grad) is implemented.
+    pytest.param(dist.Beta, osp.beta, (1., 1.), marks=pytest.mark.xfail(
+        reason='currently, variance of grad of beta sampler is large')),
+    pytest.param(dist.StudentT, osp.t, (1.,), marks=pytest.mark.xfail(
+        reason='currently, variance of grad of t sampler is large')),
+])
+def test_pathwise_gradient(jax_dist, sp_dist, params):
```

Comment:
From my understanding, `test_sample_gradient` is also a test for pathwise gradient, though I think this estimate might be more accurate (but I'm not sure since finite difference on the CDF can be numerically unstable too)?

Reply:
The CDF formula is used to compute pathwise derivative for those samplers, so it is better to test against that formula. I am not sure how precise it is when use `test_sample_gradient`, but it is better to pass these tests before passing `test_sample_gradient`.